jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -1118,13 +1118,15 @@ public static void setSecurityManager(final SecurityManager s) {
 	@SuppressWarnings("removal")
 	final SecurityManager currentSecurity = security;
 
-	if ((currentSecurity == null) && (s == null)) {
-		/* Return if the input argument is null and no security manager has been established. */
-		return;
-	}
-
 	/*[IF JAVA_SPEC_VERSION > 11]*/
```

Comment:
Just noticed the matching `/*[ENDIF] JAVA_SPEC_VERSION > 11 */` and following `/*[IF JAVA_SPEC_VERSION >= 17] */` can be removed since they are same.

Reply:
Not the same. `throwUOEFromSetSM` behaviour was introduced in Java 12. The warning messages were introduced in Java 17. The current impl correctly matches the RI.

Reply:
There is no Java 12 - 16 builds, so they are technically same.
The proposed change won't change the code generated by JPP.

Reply:
Technically Jason is correct, although there are differing opinions among committers. Java 12-16 are out of support so we only care about 11, 17, and 18+ atm (until 18 goes out of support, etc). We don't try to keep older versions successfully building, so there is no guarantee you could still build Java 12-16 using the latest code base.

Reply:
That's @keithc-ca opinion as well. In the longer term it's not going to matter. Around 2030 we should no longer need to support Java 8, 11, 17 and everything that isn't needed for Java 21+ could just be deleted.

Reply:
I would say 'practically' correct, but not 'technically' correct. As @pshipton says, my preference is not to (prematurely at least) erase technical information.